src/spline.c

```diff
@@ -93,14 +93,37 @@ static void _twin_spline_decompose(twin_path_t *path,
     _twin_path_sdraw(path, spline->a.x, spline->a.y);
 
     while (!is_flat(spline, tolerance_squared)) {
-        int shift = 1;
+        twin_dfixed_t hi = TWIN_SFIXED_ONE * TWIN_SFIXED_ONE, lo = 0;
+        twin_dfixed_t best = 0, t;
+        twin_dfixed_t best_error = 0, error;
         twin_spline_t left, right;
 
-        /* FIXME: Find the optimal shift value to decompose the spline */
-        do {
-            _de_casteljau(spline, shift, &left, &right);
-            shift++;
-        } while (!is_flat(&left, tolerance_squared));
+        while (true) {
+            t = (hi + lo) >> 1;
+            if (best == t)
+                break;
+
+            _de_casteljau(spline, t, &left, &right);
+
+            error = _twin_spline_error_squared(&left);
+            if (error < best_error)
+                break;
+
+            /* Left is close enough to the bezier curve */
```

Comment:
Capitalize "Bézier" curve.

Reply:
@weihsinyeh 
Any formula, equation, or theorem related to a person's name should have the name capitalized. For example, Euler formula, Lagrange theorem, Newton's method etc.

This is a sign of respect and basic courtesy towards these great individuals.

I have also been reminded of this many times.